### PR TITLE
service/ec2/find: update call to`DescribeRouteTables` with paginated method

### PR DESCRIPTION
**Pull request by anGie44**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #21683, #21629

Notes:
* The errors similar to
```
Error: error reading Route Table Association (rtbassoc-07d32038d3161359d): empty result
```
occur when finding the RouteTable, not the association itself. This _could_ be because we use an unpaginated `DescribeRouteTables` method that may be needed for heavily used AWS accounts. Testing this is not easily feasible, so practitioner experience will need to validate this change.

Output from acceptance testing:

```
--- PASS: TestAccEC2DefaultRouteTable_Disappears_vpc (48.86s)
--- PASS: TestAccEC2RouteTable_disappears (56.58s)
--- PASS: TestAccEC2RouteTableAssociation_disappears (69.00s)
--- PASS: TestAccEC2RouteTableDataSource_main (71.11s)
--- PASS: TestAccEC2RouteTable_basic (86.63s)
--- PASS: TestAccEC2RouteTableAssociation_Subnet_basic (94.07s)
--- PASS: TestAccEC2DefaultRouteTable_vpcEndpointAssociation (94.76s)
--- PASS: TestAccEC2RouteTableDataSource_basic (94.69s)
--- PASS: TestAccEC2RouteTableAssociation_Gateway_basic (96.05s)
--- PASS: TestAccEC2DefaultRouteTable_basic (119.19s)
--- PASS: TestAccEC2RouteTable_Disappears_subnetAssociation (74.23s)
--- SKIP: TestAccEC2RouteTable_ipv4ToCarrierGateway (0.97s)
--- SKIP: TestAccEC2RouteTable_ipv4ToLocalGateway (1.16s)
--- PASS: TestAccEC2RouteTable_requireRouteTarget (39.91s)
--- PASS: TestAccEC2DefaultRouteTable_prefixListToInternetGateway (139.22s)
--- PASS: TestAccEC2RouteTableAssociation_Subnet_changeRouteTable (140.08s)
--- PASS: TestAccEC2DefaultRouteTable_conditionalCIDRBlock (144.62s)
--- PASS: TestAccEC2RouteTableAssociation_Gateway_changeRouteTable (153.07s)
--- PASS: TestAccEC2RouteTable_ipv6ToEgressOnlyInternetGateway (111.47s)
--- PASS: TestAccEC2RouteTable_ipv4ToInternetGateway (127.00s)
--- PASS: TestAccEC2DefaultRouteTable_tags (185.06s)
--- PASS: TestAccEC2DefaultRouteTable_Route_mode (197.25s)
--- PASS: TestAccEC2RouteTable_ipv4ToVPCPeeringConnection (70.06s)
--- PASS: TestAccEC2DefaultRouteTable_swap (197.94s)
--- PASS: TestAccEC2RouteTable_vgwRoutePropagation (83.79s)
--- PASS: TestAccEC2DefaultRouteTable_revokeExistingRules (204.85s)
--- PASS: TestAccEC2RouteTable_ipv4ToInstance (138.52s)
--- PASS: TestAccEC2RouteTable_IPv6ToNetworkInterface_unattached (63.35s)
--- PASS: TestAccEC2RouteTable_conditionalCIDRBlock (48.79s)
--- PASS: TestAccEC2RouteTable_vpcClassicLink (51.10s)
--- PASS: TestAccEC2RouteTable_tags (148.57s)
--- PASS: TestAccEC2RouteTable_gatewayVPCEndpoint (59.17s)
--- PASS: TestAccEC2RouteTable_Route_mode (149.65s)
--- PASS: TestAccEC2RouteTable_prefixListToInternetGateway (51.16s)
--- PASS: TestAccEC2RouteTable_vpcMultipleCIDRs (68.34s)
--- PASS: TestAccEC2RouteTablesDataSource_basic (54.88s)
--- PASS: TestAccEC2RouteTable_IPv4ToNetworkInterfaces_unattached (116.12s)
--- PASS: TestAccEC2RouteTable_multipleRoutes (134.70s)
--- PASS: TestAccEC2DefaultRouteTable_ipv4ToVPCEndpoint (348.55s)
--- PASS: TestAccEC2RouteTable_ipv4ToNatGateway (222.10s)
--- PASS: TestAccEC2RouteTable_ipv4ToVPCEndpoint (291.82s)
--- PASS: TestAccEC2RouteTable_requireRouteDestination (328.29s)
--- PASS: TestAccEC2DefaultRouteTable_ipv4ToTransitGateway (425.79s)
--- PASS: TestAccEC2RouteTable_ipv4ToTransitGateway (389.13s)
```

### CODE REVIEW

File: internal/service/ec2/find.go [hunk 1]
```diff
@@ -447,7 +447,23 @@ func FindRouteTableByID(conn *ec2.EC2, routeTableID string) (*ec2.RouteTable, er
 }
 
 func FindRouteTable(conn *ec2.EC2, input *ec2.DescribeRouteTablesInput) (*ec2.RouteTable, error) {
```
Comment (ewbankkit) [inline internal/service/ec2/find.go:449]: For an alternative pattern take a look at `FindInternetGateway`/`FindInternetGateways` - The singular calls the plural and adds more checks.

Comment (anGie44) [reply internal/service/ec2/find.go:449]: Ooo much nicer with the return errors -- will make updates here 👍

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc PKG_NAME=internal/service/ec2 TESTARGS='-run=TestAccEC2RouteTable_basic\|TestAccEC2RouteTableAssociation_Subnet_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 20 -run=TestAccEC2RouteTable_basic\|TestAccEC2RouteTableAssociation_Subnet_basic -timeout 180m
=== RUN   TestAccEC2RouteTableAssociation_Subnet_basic
=== PAUSE TestAccEC2RouteTableAssociation_Subnet_basic
=== RUN   TestAccEC2RouteTable_basic
=== PAUSE TestAccEC2RouteTable_basic
=== CONT  TestAccEC2RouteTableAssociation_Subnet_basic
=== CONT  TestAccEC2RouteTable_basic
--- PASS: TestAccEC2RouteTable_basic (24.06s)
--- PASS: TestAccEC2RouteTableAssociation_Subnet_basic (32.69s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	36.075s
```

**Comment by jasonkinsella** [comment]
Is it possible that the lack of pagination could also cause this intermittent error?

error updating Security Group (sg-xxx): couldn't find resource

**Pull request merged**